source/Calamari/Deployment/PackageRetention/Caching/PercentFreeDiskSpacePackageCacheCleaner.cs

```diff
@@ -32,18 +35,26 @@ public PercentFreeDiskSpacePackageCleaner(ICalamariFileSystem fileSystem, ISortJ
 
         public IEnumerable<PackageIdentity> GetPackagesToRemove(IEnumerable<JournalEntry> journalEntries)
         {
+            var packageCacheRetentionStrategyString = variables.Get(PackageCacheRetentionStrategy);
+            var retentionStrategy = packageCacheRetentionStrategyString != null ? (MachinePackageCacheRetentionStrategy) Enum.Parse(typeof(MachinePackageCacheRetentionStrategy), packageCacheRetentionStrategyString) : DefaultMachinePackageCacheRetentionStrategy;
+            
+            if (OctopusFeatureToggles.ConfigurablePackageCacheRetentionFeatureToggle.IsEnabled(variables) && retentionStrategy == MachinePackageCacheRetentionStrategy.Quantities)
```

Comment:
As above, keep the strategies decoupled. `retentionStrategy != MachinePackageCacheRetentionStrategy.FreeSpace`